sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageSourceBase.java

```diff
@@ -149,11 +151,14 @@
       return ImmutableList.of();
     }
 
+    Schema sessionSchema = new Schema.Parser().parse(readSession.getAvroSchema().getSchema());
```

Comment:
Not sure how this trims the schema ? Does readSession.getAvroSchema() somehow has a smaller number of fields than targetTable.getSchema() ?

Reply:
With this change, we're no longer specifying the list of selected fields to the tables.get call from which the BigQuery schema is taken; as a result, we get the entire table schema back, so we have to trim it on the client side in the case where the client has specified selected fields. The Avro schema is returned as part of the read session and contains only the selected fields, so we use it as the basis for trimming the BigQuery schema.